titus_isolate/model/kubernetes_workload.py

```diff
@@ -63,20 +63,16 @@ def get_job_annotations_from_pod(pod: V1Pod) -> dict:
     metadata = pod.metadata
     job_id = None
 
-    if metadata.annotations.get(ANNOTATION_KEY_POD_SPEC_VERSION) is not None:
-        job_id = metadata.annotations.get(V1_ANNOTATION_KEY_JOB_ID, None)
-
-        return {
-            JOB_ID_KEY: job_id,
-        }
-
-    # pre-v1 pod spec pods
-
     if metadata.labels is not None:
         job_id = metadata.labels.get(LABEL_KEY_JOB_ID, None)
+
     if job_id is None:
         # legacy, very few pods launched a while ago
-        job_id = metadata.annotations.get(ANNOTATION_KEY_JOB_ID, 'UNKNOWN_JOB_ID')
+        job_id = metadata.annotations.get(ANNOTATION_KEY_JOB_ID, None)
+
+    if job_id is None:
+        log.error("failed to extract job ID for pod: %s", pod.metadata.name)
+        job_id = 'UNKNOWN_JOB_ID'
 
     return {
         JOB_ID_KEY: job_id,
```

Comment:
Do we need the complexity of returning annotations now? 
Can this function just be `get_job_id_from_pod` ?

Reply:
yes